### PR TITLE
Use Task in ConstraintBuilder3D

### PR DESCRIPTION
**Pull request by gaschler**

### CODE REVIEW

File: cartographer/mapping/internal/constraints/constraint_builder_3d.cc [hunk 1]
```diff
@@ -287,54 +297,44 @@ void ConstraintBuilder3D::ComputeConstraint(
...
-                    << low_resolution_score_histogram_.ToString(10);
-        }
-        constraints_.clear();
-        callback = std::move(when_done_);
-        when_done_.reset();
-      }
+    if (options_.log_matches()) {
+      LOG(INFO) << constraints_.size() << " computations resulted in "
+                << result.size() << " additional constraints.";
+      LOG(INFO) << "Score histogram:\n" << score_histogram_.ToString(10);
```
Comment (pifon2a) [inline cartographer/mapping/internal/constraints/constraint_builder_3d.cc]: Maybe, just one LOG(INFO) <<

Comment (gaschler) [reply cartographer/mapping/internal/constraints/constraint_builder_3d.cc]: Done

**Comment by gaschler** [comment]
@wally-the-cartographer merge

**Comment by wally-the-cartographer** [comment]
Merge requested by authorized user gaschler. Merge queue now has a length of 1.

**Comment by wally-the-cartographer** [comment]
Refusing to merge, since the combined checks state is failure

**Comment by cschuet** [comment]
@wally-the-cartographer merge

**Comment by wally-the-cartographer** [comment]
Merge requested by authorized user cschuet. Merge queue now has a length of 1.

**Pull request merged**